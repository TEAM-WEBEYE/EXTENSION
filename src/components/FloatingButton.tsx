import React from "react";
import styled from "@emotion/styled";

interface FloatingButtonProps {
    onClick: () => void;
    iconUrl: string;
}

const ButtonContainer = styled.div`
    position: fixed;
    top: 70px;
    right: 20px;
    width: 58px;
    height: 58px;

    display: flex;
    align-items: center;
    justify-content: center;

    cursor: pointer;
    z-index: 9999;
    transition: all 0.2s ease-in-out;

    &:hover {
        transform: scale(1.05);
    }
`;

const FloatingButton = ({ onClick, iconUrl }: FloatingButtonProps) => {
    console.log("FloatingButton iconUrl:", iconUrl);
    return (
        <ButtonContainer onClick={onClick} className="select-none">
            <img src={iconUrl} alt="Floating Button" />
        </ButtonContainer>
    );
};

export default FloatingButton;
