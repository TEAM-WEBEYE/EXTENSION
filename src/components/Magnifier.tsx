// src/components/Magnifier.tsx
import React, { useEffect, useRef, useState } from "react";

interface MagnifierProps {
    imageUrl: string;
    zoom: number;
    size: number;
    aa: boolean;
    shape: number; // percentage (e.g., 50 for circle, 0 for square)
    escOnly: boolean;
    onClose: () => void;
}

const Magnifier: React.FC<MagnifierProps> = ({
    imageUrl,
    zoom,
    size,
    aa,
    shape,
    escOnly,
    onClose,
}) => {
    const magnifierRef = useRef<HTMLDivElement>(null);
    const [visible, setVisible] = useState(false);

    const handleMouseMove = (e: MouseEvent) => {
        const magnifier = magnifierRef.current;
        if (!magnifier) return;

        const width = window.innerWidth;
        const height = window.innerHeight;

        if (
            e.clientX > 0 &&
            e.clientX < width - 1 &&
            e.clientY > 0 &&
            e.clientY < height - 4
        ) {
            setVisible(true);
            magnifier.focus();

            // 돋보기 위치 계산
            const magnifierWidth = size;
            const magnifierHeight = size;

            // 돋보기 위치 설정
            const left = e.clientX - magnifierWidth / 2;
            const top = e.clientY - magnifierHeight / 2;

            // 배경 위치 계산 (마우스 위치에 따라 배경 이미지 위치 조정)
            const xOffset = -e.clientX * zoom + magnifierWidth / 2;
            const yOffset = -e.clientY * zoom + magnifierHeight / 2;

            if (imageUrl) {
                magnifier.style.backgroundPosition = `${xOffset}px ${yOffset}px`;
            }
            magnifier.style.left = `${left}px`;
            magnifier.style.top = `${top}px`;
        } else {
            setVisible(false);
        }
    };

    const handleClose = (e: MouseEvent | KeyboardEvent) => {
        if (!escOnly || (e instanceof KeyboardEvent && e.key === "Escape")) {
            onClose();
        }
    };

    useEffect(() => {
        document.addEventListener("mousemove", handleMouseMove);
        document.addEventListener("wheel", handleClose);
        document.addEventListener("click", handleClose);
        document.addEventListener("keydown", handleClose);

        return () => {
            document.removeEventListener("mousemove", handleMouseMove);
            document.removeEventListener("wheel", handleClose);
            document.removeEventListener("click", handleClose);
            document.removeEventListener("keydown", handleClose);
        };
    }, [zoom, escOnly]);

    useEffect(() => {
        // 스크린샷이 로드되었는지 확인
        if (imageUrl) {
            const img = new Image();
            img.src = imageUrl;
            img.onload = () => {
                console.log("Screenshot loaded successfully");
                // 스크린샷이 로드되면 돋보기 표시
                setVisible(true);
            };
            img.onerror = (error) => {
                console.error("Error loading screenshot:", error);
            };
        }
    }, [imageUrl]);

    return (
        <div
            ref={magnifierRef}
            className={`fixed z-[9999] bg-no-repeat pointer-events-none transition-opacity duration-100 overflow-hidden ${
                visible ? "opacity-100" : "opacity-0"
            }`}
            style={{
                backgroundImage: imageUrl ? `url(${imageUrl})` : "none",
                backgroundSize: `${window.innerWidth * zoom}px ${
                    window.innerHeight * zoom
                }px`,
                imageRendering: aa ? "auto" : "pixelated",
                width: `${size}px`,
                height: `${size}px`,
                borderRadius: `${shape}%`,
                boxShadow: `0 0 0 7px rgba(255,255,255,0.85),
                    0 0 7px 7px rgba(0,0,0,0.25),
                    inset 0 0 40px 2px rgba(0,0,0,0.25)`,
            }}
            tabIndex={0}
        />
    );
};

export default Magnifier;
