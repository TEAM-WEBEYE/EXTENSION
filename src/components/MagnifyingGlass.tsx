/* eslint-disable react/prop-types */
import { useEffect, useRef, useState } from "react";
import React = require("react");

interface MagnifyingGlassProps {
    snapshotUrl: string;
    magnifierStrength: number;
    magnifierSize: number;
    magnifierAA: boolean;
    magnifierShape: number;
    pageZoom: number;
    osCompensation: number;
    escOnly: boolean;
}

const MagnifyingGlass: React.FC<MagnifyingGlassProps> = ({
    snapshotUrl,
    magnifierStrength,
    magnifierSize,
    magnifierAA,
    magnifierShape,
    pageZoom,
    osCompensation,
    escOnly,
}) => {
    const magnifierRef = useRef<HTMLDivElement>(null);
    const containerRef = useRef<HTMLDivElement>(null);

    const [visible, setVisible] = useState(false);

    const zoom = pageZoom * (osCompensation / 100);
    const scaledSize = magnifierSize / zoom;
    const sizeStyle = `${scaledSize / magnifierStrength}px`;
    const boxShadow = `
    0 0 0 ${7 / magnifierStrength}px rgba(255, 255, 255, 0.85),
    0 0 ${7 / magnifierStrength}px ${
        7 / magnifierStrength
    }px rgba(0, 0, 0, 0.25),
    inset 0 0 ${40 / magnifierStrength}px ${
        2 / magnifierStrength
    }px rgba(0, 0, 0, 0.25)
  `;

    useEffect(() => {
        const handleMouseMove = (e: MouseEvent) => {
            const container = containerRef.current;
            const magnifier = magnifierRef.current;
            if (!container || !magnifier) return;

            const { clientX, clientY } = e;
            const { offsetWidth, offsetHeight } = container;

            if (
                clientX > 0 &&
                clientY > 0 &&
                clientX < offsetWidth - 1 &&
                clientY < offsetHeight - 4
            ) {
                setVisible(true);
                magnifier.focus();
            } else {
                setVisible(false);
            }

            if (magnifier.style.display !== "none") {
                const xOffset =
                    -1 * (clientX - magnifier.offsetWidth / 2) * zoom;
                const yOffset =
                    -1 * (clientY - magnifier.offsetHeight / 2) * zoom;
                const bgPosition = `${xOffset}px ${yOffset}px`;

                magnifier.style.left = `${
                    clientX - magnifier.offsetWidth / 2
                }px`;
                magnifier.style.top = `${
                    clientY - magnifier.offsetHeight / 2
                }px`;
                magnifier.style.backgroundPosition = bgPosition;
            }
        };

        const handleClose = (e: MouseEvent | KeyboardEvent) => {
            if (
                !escOnly ||
                ("key" in e && (e as KeyboardEvent).key === "Escape")
            ) {
                containerRef.current?.remove();
            }
        };

        document.addEventListener("mousemove", handleMouseMove);
        document.addEventListener("click", handleClose);
        document.addEventListener("wheel", handleClose);
        document.addEventListener("keydown", handleClose);

        return () => {
            document.removeEventListener("mousemove", handleMouseMove);
            document.removeEventListener("click", handleClose);
            document.removeEventListener("wheel", handleClose);
            document.removeEventListener("keydown", handleClose);
        };
    }, [zoom, escOnly]);

    return (
        <div
            ref={containerRef}
            className="fixed top-0 left-0 w-screen h-screen z-[9999]"
        >
            <div
                ref={magnifierRef}
                tabIndex={0}
                className="absolute pointer-events-none bg-no-repeat bg-cover"
                style={{
                    display: visible ? "block" : "none",
                    backgroundImage: `url(${snapshotUrl})`,
                    width: sizeStyle,
                    height: sizeStyle,
                    transform: `scale(${magnifierStrength})`,
                    borderRadius: `${magnifierShape}%`,
                    imageRendering: magnifierAA ? "auto" : "pixelated",
                    boxShadow,
                }}
            />
        </div>
    );
};

export default MagnifyingGlass;
