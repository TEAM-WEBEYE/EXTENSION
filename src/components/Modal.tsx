import React from "react";
import styled from "@emotion/styled";

interface ModalProps {
    isOpen: boolean;
    onClose: () => void;
    children: React.ReactNode;
}

const ModalOverlay = styled.div<{ isOpen: boolean }>`
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(0, 0, 0, 0.5);
    display: ${(props) => (props.isOpen ? "flex" : "none")};
    align-items: center;
    justify-content: center;
    z-index: 10000;
`;

const ModalContainer = styled.div`
    background-color: white;
    padding: 1.5rem;
    border-radius: 0.5rem;
    width: 90%;
    max-width: 500px;
    max-height: 80vh;
    overflow-y: auto;
    position: relative;
    box-shadow: 0 10px 15px -3px rgba(0, 0, 0, 0.1),
        0 4px 6px -2px rgba(0, 0, 0, 0.05);
`;

const CloseButton = styled.button`
    position: absolute;
    top: 0.75rem;
    right: 0.75rem;
    background: none;
    border: none;
    font-size: 1.5rem;
    cursor: pointer;
    color: #6b7280; /* gray-500 */

    &:hover {
        color: #1f2937; /* gray-800 */
    }
`;

const Modal: React.FC<ModalProps> = ({ isOpen, onClose, children }) => {
    // 모달 외부 클릭 시 닫기
    const handleOverlayClick = (e: React.MouseEvent<HTMLDivElement>) => {
        if (e.target === e.currentTarget) {
            onClose();
        }
    };

    return (
        <ModalOverlay isOpen={isOpen} onClick={handleOverlayClick}>
            <ModalContainer className="font-sans">
                <CloseButton onClick={onClose}>&times;</CloseButton>
                {children}
            </ModalContainer>
        </ModalOverlay>
    );
};

export default Modal;
