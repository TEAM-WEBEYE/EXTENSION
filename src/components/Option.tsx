import React from "react";
import { useEffect, useState } from "react";

export default function Options() {
    const [magnifierStrength, setMagnifierStrength] = useState(2);
    const [magnifierSize, setMagnifierSize] = useState(425);
    const [magnifierAA, setMagnifierAA] = useState(true);
    const [magnifierCM, setMagnifierCM] = useState(false);
    const [magnifierShape, setMagnifierShape] = useState(100);
    const [osFactor, setOsFactor] = useState(100);
    const [escLimit, setEscLimit] = useState(false);
    const [status, setStatus] = useState("");

    useEffect(() => {
        chrome.storage.sync.get(
            {
                magnifierStrength: 2,
                magnifierSize: 425,
                magnifierAA: true,
                magnifierCM: false,
                magnifierShape: 100,
                osFactor: 100,
                escLimit: false,
            },
            (items) => {
                setMagnifierStrength(items.magnifierStrength);
                setMagnifierSize(items.magnifierSize);
                setMagnifierAA(items.magnifierAA);
                setMagnifierCM(items.magnifierCM);
                setMagnifierShape(items.magnifierShape);
                setOsFactor(items.osFactor);
                setEscLimit(items.escLimit);
            },
        );
    }, []);

    const saveOptions = () => {
        chrome.storage.sync.set(
            {
                magnifierStrength,
                magnifierSize,
                magnifierAA,
                magnifierCM,
                magnifierShape,
                osFactor,
                escLimit,
            },
            () => {
                if (magnifierStrength <= 0 || magnifierSize <= 0) {
                    setStatus(
                        "Non-positive number detected, the magnifier might not behave as expected.",
                    );
                } else {
                    setStatus("Options saved.");
                }
                setTimeout(() => setStatus(""), 1500);
            },
        );
    };

    const resetOptions = () => {
        chrome.storage.sync.clear(() => {
            chrome.storage.sync.set(
                {
                    magnifierStrength: 2,
                    magnifierSize: 425,
                    magnifierAA: true,
                    magnifierCM: false,
                    magnifierShape: 100,
                    osFactor: 100,
                    escLimit: false,
                },
                () => {
                    setMagnifierStrength(2);
                    setMagnifierSize(425);
                    setMagnifierAA(true);
                    setMagnifierCM(false);
                    setMagnifierShape(100);
                    setOsFactor(100);
                    setEscLimit(false);
                    setStatus("Options reset.");
                    setTimeout(() => setStatus(""), 1200);
                },
            );
        });
    };

    return (
        <div className="p-4 max-w-xl mx-auto text-sm space-y-4">
            <div className="font-bold text-lg">Magnifier Options</div>

            <div>
                <label>Strength:</label>
                <input
                    type="number"
                    value={magnifierStrength}
                    onChange={(e) =>
                        setMagnifierStrength(parseFloat(e.target.value))
                    }
                    className="border rounded px-2 py-1 ml-2 w-20"
                />
            </div>

            <div>
                <label>Size:</label>
                <input
                    type="number"
                    value={magnifierSize}
                    onChange={(e) => setMagnifierSize(parseInt(e.target.value))}
                    className="border rounded px-2 py-1 ml-2 w-24"
                />
            </div>

            <div className="flex items-center space-x-2">
                <input
                    type="checkbox"
                    checked={magnifierAA}
                    onChange={(e) => setMagnifierAA(e.target.checked)}
                />
                <label>Anti-aliasing</label>
            </div>

            <div className="flex items-center space-x-2">
                <input
                    type="checkbox"
                    checked={magnifierCM}
                    onChange={(e) => setMagnifierCM(e.target.checked)}
                />
                <label>Compatibility Mode</label>
            </div>

            <div>
                <label>Shape (% circle):</label>
                <input
                    type="number"
                    value={magnifierShape}
                    onChange={(e) =>
                        setMagnifierShape(parseInt(e.target.value))
                    }
                    className="border rounded px-2 py-1 ml-2 w-20"
                />
            </div>

            <div>
                <label>OS Zoom Compensation (%):</label>
                <input
                    type="number"
                    value={osFactor}
                    onChange={(e) => setOsFactor(parseInt(e.target.value))}
                    className="border rounded px-2 py-1 ml-2 w-24"
                />
            </div>

            <div className="flex items-center space-x-2">
                <input
                    type="checkbox"
                    checked={escLimit}
                    onChange={(e) => setEscLimit(e.target.checked)}
                />
                <label>ESC Key Only to Exit</label>
            </div>

            <div className="flex space-x-2">
                <button
                    onClick={saveOptions}
                    className="bg-blue-500 text-white px-4 py-1 rounded hover:bg-blue-600"
                >
                    Save
                </button>
                <button
                    onClick={resetOptions}
                    className="bg-gray-300 px-4 py-1 rounded hover:bg-gray-400"
                >
                    Reset
                </button>
            </div>

            {status && (
                <div className="text-green-600 font-medium">{status}</div>
            )}
        </div>
    );
}
