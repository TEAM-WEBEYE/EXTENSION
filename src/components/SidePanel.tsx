import React from "react";
import styled from "@emotion/styled";

interface SidePanelProps {
    isOpen: boolean;
    onClose: () => void;
    children: React.ReactNode;
}

const SidePanelOverlay = styled.div<{ isOpen: boolean }>`
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    display: ${(props) => (props.isOpen ? "block" : "none")};
    z-index: 9998;
`;

const SidePanelContainer = styled.div<{ isOpen: boolean }>`
    position: fixed;
    top: 70px;
    right: 20px;
    width: 460px;
    height: 726px;
    background-color: white;
    box-shadow: 0px 0px 4px 0px rgba(0, 0, 0, 0.25);
    border-radius: 30px;
    z-index: 9999;
    display: ${(props) => (props.isOpen ? "block" : "none")};
    transition: transform 0.3s ease;
    overflow-y: auto;
`;

const PanelHeader = styled.div`
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 16px;
    border-bottom: 1px solid #eaeaea;
`;

const CloseButton = styled.button`
    background: none;
    border: none;
    cursor: pointer;
    font-size: 20px;
    color: #6b7280;
    &:hover {
        color: #1f2937;
    }
`;

const PanelContent = styled.div`
    padding: 16px;
`;

const SidePanel: React.FC<SidePanelProps> = ({ isOpen, onClose, children }) => {
    // 패널 외부 클릭 시 닫기
    const handleOverlayClick = (e: React.MouseEvent<HTMLDivElement>) => {
        if (e.target === e.currentTarget) {
            onClose();
        }
    };

    return (
        <>
            <SidePanelOverlay isOpen={isOpen} onClick={handleOverlayClick} />
            <SidePanelContainer isOpen={isOpen}>
                <PanelHeader>
                    <h3 className="text-lg font-semibold">익스텐션 패널</h3>
                    <CloseButton onClick={onClose}>×</CloseButton>
                </PanelHeader>
                <PanelContent>{children}</PanelContent>
            </SidePanelContainer>
        </>
    );
};

export default SidePanel;
