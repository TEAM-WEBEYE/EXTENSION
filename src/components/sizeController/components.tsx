import React from "react";
import { getExtensionUrl } from "@src/background/utils/getExtensionUrl";

interface SizeControllerProps {
    type: "minus" | "plus";
}

export function SizeController({ type }: SizeControllerProps): JSX.Element {
    return (
        <div className="px-[54px] py-[18px] inline-block rounded-[14px] bg-purple-default hover:bg-purple-dark">
            <img
                src={getExtensionUrl(`${type}.png`)}
                alt={type}
                className="w-[36px] h-[36px]"
            />
        </div>
    );
}
