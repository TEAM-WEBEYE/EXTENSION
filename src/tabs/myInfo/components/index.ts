export { Onboarding } from "./OnBoarding";
export { InfoForm } from "./InfoForm";
