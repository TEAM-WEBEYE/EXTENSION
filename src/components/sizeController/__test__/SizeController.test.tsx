import * as React from "react";
import { SizeController } from "../components";
import renderer from "react-test-renderer";

describe("SizeController", () => {
    it("renders minus button correctly", () => {
        const tree = renderer.create(<SizeController type="minus" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders plus button correctly", () => {
        const tree = renderer.create(<SizeController type="plus" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
