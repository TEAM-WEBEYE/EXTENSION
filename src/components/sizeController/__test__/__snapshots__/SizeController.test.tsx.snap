// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SizeController + 버튼 렌더링 1`] = `
<div
  className="px-[54px] py-[18px] inline-block rounded-[14px] bg-purple-default active:bg-purple-dark hover:bg-purple-light"
  onClick={[MockFunction]}
>
  <img
    alt="plus"
    className="w-[36px] h-[36px]"
    draggable="false"
    src="/images/plus.png"
  />
</div>
`;

exports[`SizeController - 버튼 렌더링 1`] = `
<div
  className="px-[54px] py-[18px] inline-block rounded-[14px] bg-purple-default active:bg-purple-dark hover:bg-purple-light"
  onClick={[MockFunction]}
>
  <img
    alt="minus"
    className="w-[36px] h-[36px]"
    draggable="false"
    src="/images/minus.png"
  />
</div>
`;
